[share_plus] minSdkVersion 16 & graceful fallback (#800)
fluttercommunity/plus_plugins: packages/share_plus/share_plus_platform_interface/CHANGELOG.md

## 3.0.2

> - Gracefully fall back to normal share methods on not supported platforms

Reviewer: ```suggestion
- Gracefully fall back from `shareWithResult()` to regular `share()` methods on unsupported platforms
```
